### ISSUE TITLE
Improve accessibilty of statements index

### ISSUE THREAD

**Issue by peteryates** [DFE-Digital/register-early-career-teachers-public#713]
The statements index page currently has a load of links that just say "View". A screen reader will just read this out as "View, View, View, View".

![Image](https://github.com/user-attachments/assets/215ed1c9-5bd6-4576-a4b4-03dc0e1d0763)

This should be more informative. There needs to be some [visually hidden text](https://govuk-components.netlify.app/helpers/visually-hidden-text/) that explains what the person will be viewing if they click the link.

At a minimum it should be "View statement 123" but it's likely going to be more useful if it says "View statement for [Lead Provider Name] in [Month and Year]".
